Design Token - style builder (#2251)
AmadeusITGroup/otter: packages/@o3r/design/README.md

Otter Design module provides a set of builders based on [Angular builders](https://angular.io/guide/cli-builder).

### generate-style

> 

Author: Actually I followed a strategy common on the web which is to remove from highlighted doc the deprecated items but I can add a line regarding the deprecated builder indeed.

| **variableType**            | *null*             | Type of the variables to generate for a Design Token.                                                                                        |
| **output**                  | *null*             | Output file where the CSS will be generated. <br /> The path specified in `o3rTargetFile` will be ignored if this option is specified        |
| **defaultStyleFile**        | `'src/theme.scss'` | File path to generate the variable if not determined by the specifications                                                                   |
| **metadataOutput**          | *null*             | Path to generate the metadata for the CMS. <br /> The metadata will be generated only if the file path is specified.                         |
| **metadataIgnorePrivate**   | false              | Ignore private variables in the metadata generation.                                                                                         |
| **rootPath**                | *null*             | Root path of files where the CSS will be generated.                                                                                          |

> | **failOnDuplicate**         | false              | Determine if the process should stop in case of Token duplication.                                                                           |

Reviewer: ```suggestion
| **failOnDuplicate**         | false              | Determine if the process should stop in case of token duplication.                                                                           |
```

Author: I think `Design Token`, `Token` and `Metadata` should remain with upper case.
Design Token is a standard name, same for Metadata that stands for the EAM Metadata protocol.
(We did already that in the current code)

| **templateFile**            | *null*             | Path to a template file (or a list of template files) to apply as default configuration to a Design Token extension.                         |
| **prefix**                  | *null*             | Prefix to append to generated variables.                                                                                                     |
| **prefixPrivate**           | *null*             | Prefix to append to generated private variables.                                                                                             |
| **renderPrivateVariableTo** | *null*             | Generate the private variable in the given language (if not specified, in case of CSS, the variable will be ignored in the generated style). |

> | **watch**                   | false              | Enable Watch mode.                                                                                                                           |

Reviewer: ```suggestion
| **watch**                   | false              | Enable watch mode.                                                                                                                           |
```

| **o3rMetadata**       | `css`, `sass`, `json-schema`, `metadata`, `design-token` | Additional information to provide to the metadata if generated                                                                                                                                                                                                                   |
| **o3rUnit**           | `css`, `sass`, `metadata`, `design-token`                | Convert a numeric value from the specified unit to the new unit. It will add a unit to the token with type "number" for which the unit is not specified.<br />In case of complex type (such as shadow, transition, etc...), the unit will be applied to all numeric types in it. |
| **o3rRatio**          | `css`, `sass`, `metadata`, `design-token`                | Ratio to apply to previous value. The ratio will be applied only on token with "number" type or on the first numbers determined in "string" like types.<br />In case of complex type (such as shadow, transition, etc...), the ratio will be applied to all numeric types in it. |

> | **o3rExpectOverride** | `css`, `sass`                                            | Indicate that the token expect to be overridden by external rules.                                                                                                                                                                                                               |

Reviewer 2: ```suggestion
| **o3rExpectOverride** | `css`, `sass`                                            | Indicate that the token is expected to be overridden by external rules.                                                                                                                                                                                                               |
```
